chloggen panics on execution

chloggen panics when called to generate the changelog with this build:
https://github.com/open-telemetry/opentelemetry-collector-contrib/actions/runs/3953803247/jobs/6770466495


See this stacktrace:
```
panic: open go.opentelemetry.io/build-tools/chloggen@v0.3.0/internal/chlog/summary.tmpl: no such file or directory

goroutine 1 [running]:
text/template.Must(...)
	text/template/helper.go:26
go.opentelemetry.io/build-tools/chloggen/internal/chlog.summary.String({{0x5ebe97, 0x5}, {0xc000146400, 0x12, 0x20}, {0xc000197d70, 0x1, 0x1}, {0x0, 0x0, ...}, ...})
	go.opentelemetry.io/build-tools/chloggen@v0.3.0/internal/chlog/summary.go:66 +0x311
go.opentelemetry.io/build-tools/chloggen/internal/chlog.GenerateSummary({0x5ebe97?, 0x51?}, {0xc00014[8](https://github.com/open-telemetry/opentelemetry-collector-contrib/actions/runs/3953601887/jobs/6770052166#step:8:9)000, 0x2d, 0xc000026240?})
	go.opentelemetry.io/build-tools/chloggen@v0.3.0/internal/chlog/summary.go:60 +0x73f
go.opentelemetry.io/build-tools/chloggen/cmd.update({{0xc000026004, 0x51}, {0xc000026180, 0x5e}, {0xc000026240, 0x5b}, {0xc000020540, 0x6[9](https://github.com/open-telemetry/opentelemetry-collector-contrib/actions/runs/3953601887/jobs/6770052166#step:8:10)}}, {0x5ebe97, 0x5}, ...)
	go.opentelemetry.io/build-tools/chloggen@v0.3.0/cmd/update.go:56 +0xc7
go.opentelemetry.io/build-tools/chloggen/cmd.glob..func2(0x758e20?, {0x5ebbc7?, 0x1?, 0x1?})
	go.opentelemetry.io/build-tools/chloggen@v0.3.0/cmd/update.go:42 +0x70
github.com/spf13/cobra.(*Command).execute(0x758e20, {0xc000030880, 0x1, 0x1})
	github.com/spf13/cobra@v1.6.1/command.go:916 +0x862
github.com/spf13/cobra.(*Command).ExecuteC(0x758b40)
	github.com/spf13/cobra@v1.6.1/command.go:[10](https://github.com/open-telemetry/opentelemetry-collector-contrib/actions/runs/3953601887/jobs/6770052166#step:8:11)44 +0x3bd
github.com/spf[13](https://github.com/open-telemetry/opentelemetry-collector-contrib/actions/runs/3953601887/jobs/6770052166#step:8:14)/cobra.(*Command).Execute(...)
	github.com/spf13/cobra@v1.6.1/command.go:968
go.opentelemetry.io/build-tools/chloggen/cmd.Execute()
	go.opentelemetry.io/build-tools/chloggen@v0.3.0/cmd/root.go:35 +0x25
main.main()
	go.opentelemetry.io/build-tools/chloggen@v0.3.0/main.go:22 +0x[17](https://github.com/open-telemetry/opentelemetry-collector-contrib/actions/runs/3953601887/jobs/6770052166#step:8:18)
make: *** [Makefile:[24](https://github.com/open-telemetry/opentelemetry-collector-contrib/actions/runs/3953601887/jobs/6770052166#step:8:25)9: chlog-preview] Error 2
Error: Process completed with exit code 2.
```

--- a/.chloggen/use-go-embed-for-chloggen-template.yaml
+++ b/.chloggen/use-go-embed-for-chloggen-template.yaml
@@ -1,0 +1,16 @@
+# One of 'breaking', 'deprecation', 'new_component', 'enhancement', 'bug_fix'
+change_type: enhancement
+
+# The name of the component, or a single word describing the area of concern, (e.g. crosslink)
+component: chloggen
+
+# A brief description of the change.  Surround your text with quotes ("") if it needs to start with a backtick (`).
+note: Use go embed to embed the template file in chloggen
+
+# One or more tracking issues related to the change
+issues: [224]
+
+# (Optional) One or more lines of additional information to render under the primary note.
+# These lines will be padded with 2 spaces and then inserted directly into the document.
+# Use pipe (|) for multiline entries.
+subtext:

--- a/chloggen/internal/chlog/context.go
+++ b/chloggen/internal/chlog/context.go
@@ -18,7 +18,6 @@ import (
 	"fmt"
 	"os"
 	"path/filepath"
-	"runtime"
 )
 
 const (
@@ -64,13 +63,4 @@ func RepoRoot() string {
 		fmt.Println("FAIL: Could not determine current working directory")
 	}
 	return dir
-}
-
-func moduleDir() string {
-	_, filename, _, ok := runtime.Caller(0)
-	if !ok {
-		// This is not expected, but just in case
-		fmt.Println("FAIL: Could not determine module directory")
-	}
-	return filepath.Dir(filename)
 }

--- a/chloggen/internal/chlog/summary.go
+++ b/chloggen/internal/chlog/summary.go
@@ -23,7 +23,7 @@ import (
 )
 
 //go:embed summary.tmpl
-var tmpl embed.FS
+var tmpl []byte
 
 type summary struct {
 	Version         string

--- a/chloggen/internal/chlog/summary.go
+++ b/chloggen/internal/chlog/summary.go
@@ -16,11 +16,14 @@ package chlog
 
 import (
 	"bytes"
+	"embed"
 	"fmt"
-	"path/filepath"
 	"sort"
 	"text/template"
 )
+
+//go:embed summary.tmpl
+var tmpl embed.FS
 
 type summary struct {
 	Version         string
@@ -61,13 +64,11 @@ func GenerateSummary(version string, entries []*Entry) (string, error) {
 }
 
 func (s summary) String() (string, error) {
-	summaryTmpl := filepath.Join(moduleDir(), "summary.tmpl")
-
 	tmpl := template.Must(
 		template.
 			New("summary.tmpl").
 			Option("missingkey=error").
-			ParseFiles(summaryTmpl))
+			ParseFS(tmpl, "summary.tmpl"))
 
 	buf := bytes.Buffer{}
 	if err := tmpl.Execute(&buf, s); err != nil {

--- a/chloggen/internal/chlog/summary.go
+++ b/chloggen/internal/chlog/summary.go
@@ -68,7 +68,7 @@ func (s summary) String() (string, error) {
 		template.
 			New("summary.tmpl").
 			Option("missingkey=error").
-			ParseFS(tmpl, "summary.tmpl"))
+			Parse(string(tmpl))
 
 	buf := bytes.Buffer{}
 	if err := tmpl.Execute(&buf, s); err != nil {

--- a/chloggen/internal/chlog/summary.go
+++ b/chloggen/internal/chlog/summary.go
@@ -16,7 +16,7 @@ package chlog
 
 import (
 	"bytes"
-	"embed"
+	_ "embed"
 	"fmt"
 	"sort"
 	"text/template"
@@ -68,7 +68,7 @@ func (s summary) String() (string, error) {
 		template.
 			New("summary.tmpl").
 			Option("missingkey=error").
-			Parse(string(tmpl))
+			Parse(string(tmpl)))
 
 	buf := bytes.Buffer{}
 	if err := tmpl.Execute(&buf, s); err != nil {

--- a/chloggen/internal/chlog/summary_test.go
+++ b/chloggen/internal/chlog/summary_test.go
@@ -1,0 +1,59 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package chlog
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func Test_SummaryString(t *testing.T) {
+	s := summary{
+		Version:         "1.0",
+		BreakingChanges: []string{"foo", "bar"},
+		Deprecations:    []string{"foo", "bar"},
+		NewComponents:   []string{"foo", "bar", "new component"},
+		Enhancements:    []string{},
+		BugFixes:        []string{"foo", "bar", "foobar"},
+	}
+	result, err := s.String()
+	assert.NoError(t, err)
+	assert.Equal(t, `
+## 1.0
+
+### 🛑 Breaking changes 🛑
+
+foo
+bar
+
+### 🚩 Deprecations 🚩
+
+foo
+bar
+
+### 🚀 New components 🚀
+
+foo
+bar
+new component
+
+### 🧰 Bug fixes 🧰
+
+foo
+bar
+foobar
+`, result)
+}

